esrally/storage/_adapter.py

```diff
@@ -114,33 +112,34 @@ def head(self, url: str) -> Head:
         """
 
     @abstractmethod
-    def get(self, url: str, stream: Writable, ranges: RangeSet = NO_RANGE) -> Head:
+    def get(self, url: str, stream: Writable, head: Head | None = None) -> Head:
         """It downloads a remote bucket object to a local file path.
 
         :param url: it represents the URL of the remote file object.
         :param stream: it represents the local file stream where to write data to.
-        :param ranges: it represents the portion of the file to transfer (it must be empty or a continuous range).
+        :param head: it allows to specify optional parameters:
+            - range: portion of the file to transfer (it must be empty or a continuous range).
+            - document_length: the number of bytes to transfer.
+            - crc32c the CRC32C checksum of the file.
+            - date: the date the file has been modified.
         :raises ServiceUnavailableError: in case on temporary service failure.
         """
 
 
 ADAPTER_CLASS_NAMES = ",".join(
```

Comment:
Hmm, I'm not sure I like this interface. Could we annotate it as `Final[list[str]]` and just use a list? I don't know if it's the best idea to join and split later.